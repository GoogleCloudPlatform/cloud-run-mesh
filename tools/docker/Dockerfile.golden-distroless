# Dockerfile using an external build (similar with Istio)

ARG BASE=gcr.io/istio-testing/proxyv2:latest-distroless

FROM ${BASE}

COPY krun /usr/local/bin/krun
COPY bootstrap_template.yaml /td_resources/
COPY iptables.sh /td_resources/

# TODO: allow workdir to be determined by app, cd to / for creating the files if root.
WORKDIR /
USER 0
ENTRYPOINT ["/usr/local/bin/krun"]
