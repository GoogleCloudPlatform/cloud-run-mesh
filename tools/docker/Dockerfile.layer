
ARG BASE=gcr.io/istio-testing/proxyv2:latest

FROM scratch

COPY krun /usr/local/bin/krun

WORKDIR /

ENTRYPOINT ["/usr/local/bin/krun"]
