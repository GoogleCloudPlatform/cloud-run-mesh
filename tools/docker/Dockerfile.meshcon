# Dockerfile using an external build (similar with Istio)

ARG BASE=gcr.io/istio-testing/proxyv2:latest

FROM ${BASE}

COPY hgate /usr/local/bin/hgate

WORKDIR /

ENTRYPOINT ["/usr/local/bin/hgate"]
