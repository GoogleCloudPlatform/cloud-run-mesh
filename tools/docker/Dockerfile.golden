# Dockerfile using an external build (similar with Istio)

ARG BASE=gcr.io/istio-testing/proxyv2:latest

FROM ${BASE}

COPY krun /usr/local/bin/krun
COPY bootstrap_template.yaml /td_resources/
COPY iptables.sh /td_resources/

WORKDIR /

# Add a version variant, since this is different enough in packaging.
ENV ISTIO_META_ISTIO_VERSION 1.13-master-cloudrun

ENTRYPOINT ["/usr/local/bin/krun"]
